pkg/conference/participant/tracker.go

```diff
@@ -200,19 +249,22 @@ func (t *Tracker) Unsubscribe(participantID ID, tracks []TrackID) {
 }
 
 // Processes an RTP packet received on a given track.
-func (t *Tracker) ProcessRTP(info common.TrackInfo, packet *rtp.Packet) {
+func (t *Tracker) ProcessRTP(info common.TrackInfo, simulcast common.SimulcastLayer, packet *rtp.Packet) {
 	for participantID, subscription := range t.subscribers[info.TrackID] {
-		if subscription.TrackInfo().Layer == info.Layer {
+		if subscription.Simulcast() == simulcast {
 			if err := subscription.WriteRTP(packet); err != nil {
-				participant := t.GetParticipant(participantID)
-				participant.Logger.Errorf("Error writing RTP packet to %s: %s", info.TrackID, err)
+				if participant := t.GetParticipant(participantID); participant != nil {
+					participant.Logger.Errorf("Error writing RTP to %s (%s): %s", info.TrackID, simulcast, err)
+					continue
+				}
+				logrus.Errorf("Bug: subscription without subscriber")
 			}
 		}
 	}
 }
 
 // Processes RTCP packets received on a given track.
-func (t *Tracker) ProcessKeyFrameRequest(info common.TrackInfo) error {
+func (t *Tracker) ProcessKeyFrameRequest(info common.TrackInfo, simulcast common.SimulcastLayer) error {
 	const sendKeyFrameInterval = 500 * time.Millisecond
```

Comment:
This looks unused now?

Reply:
Oh, good catch, thanks! Golang linter unfortunately did not capture that.